client/src/modules/events/pages/eventsPage.tsx

```diff
@@ -1,17 +1,76 @@
-import { Heading, VStack, Stack } from '@chakra-ui/layout';
+import { Heading, VStack, Stack, Center } from '@chakra-ui/layout';
 import { NextPage } from 'next';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Button, Box, Flex } from '@chakra-ui/react';
 import { EventCard } from 'components/EventCard';
-import { useMinEventsQuery } from 'generated/graphql';
+import { usePaginatedEventsWithTotalQuery } from 'generated/graphql';
 
+export default function Pagination({
+  currentPage = 1,
+  setCurrentPage,
+  pageSize,
+  records,
+}: {
+  currentPage: number;
+  setCurrentPage: (page: number) => void;
+  pageSize: number;
+  records: number;
+}) {
+  const totalPages = Math.ceil(records / pageSize);
+
+  return (
+    <Flex>
+      <Center>
+        <Button
+          colorScheme="blue"
+          data-testid="pagination-back"
+          disabled={!(currentPage > 1)}
+          onClick={() => currentPage > 1 && setCurrentPage(currentPage - 1)}
+        >
+          &lt;
+        </Button>
+        <Box p="5">Page</Box>
+        <Box p="5" data-testid="current-page">
+          {currentPage}
+        </Box>
+        <Box p="5">of</Box>
+        <Box p="5" data-testid="total-pages">
+          {totalPages}
+        </Box>
+        <Button
+          colorScheme="blue"
+          data-testid="pagination-forward"
+          disabled={!(currentPage < totalPages)}
+          onClick={() =>
+            currentPage < totalPages && setCurrentPage(currentPage + 1)
+          }
+        >
+          &gt;
+        </Button>
+      </Center>
+    </Flex>
+  );
+}
+const pageSize = 5;
 export const EventsPage: NextPage = () => {
-  const { loading, error, data } = useMinEventsQuery();
+  const [currentPage, setCurrentPage] = useState(1);
+  const { loading, error, data, fetchMore } = usePaginatedEventsWithTotalQuery({
+    variables: { offset: (currentPage - 1) * pageSize, limit: pageSize },
+  });
 
+  useEffect(() => {
+    fetchMore({
+      variables: {
+        offset: data?.paginatedEventsWithTotal?.events?.length || 0,
+        limit: pageSize,
+      },
+    });
+  }, [currentPage]);
   if (loading) {
     return <h1>Loading...</h1>;
   }
 
-  if (error || !data?.events) {
+  if (error || !data?.paginatedEventsWithTotal?.events?.length) {
```

Comment:
```suggestion
  if (error || !data?.paginatedEventsWithTotal.events.length) {
```
Once we know `data` exists, the rest have to be defined.